packages/system-data/src/fields/settings.yaml

```diff
@@ -200,6 +200,60 @@ fields:
       placeholder: $t:unlimited
     width: half
 
+  - field: public_registration_divider
+    interface: presentation-divider
+    options:
+      icon: person_add
+      title: $t:fields.directus_settings.public_registration
+    special:
+      - alias
+      - no-data
+    width: full
+
+  - field: public_registration
+    interface: boolean
+    note: $t:fields.directus_settings.public_registration_note
+    width: half
+    options:
+      label: $t:enabled
+    special:
+      - cast-boolean
+
+  - field: public_registration_role
+    interface: select-dropdown-m2o
+    note: $t:fields.directus_settings.public_registration_role_note
+    options:
```

Comment:
Do we have the possibility to add validation rules here? That the user can not leave this `null` if the public registration is turned on? Don't know if that is possible in there definition files, but would be a nice to have.